### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?

The Ruby on Rails capstone project is about building a mobile web application where we can manage our budget: we have a list of transactions associated with a category, so that we can see how much money we spent and on what.

Description of the task to be completed 👎

in this PR I will do bellow tasks 👎

Interactions

- [ ] Splash screen
   - A simple page with the name of your app, and links to the sign up and log in pages.

- [ ] Sign up and log in pages
   - The user should be able to register in the app with full name, email and password (all mandatory).

- [ ] The user can log into the app using email and password.

- [ ] If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [ ] Home page (categories page)
      - When the user logs in, they are presented with the categories page. 
      - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
      - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
      - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [ ] Transactions page
      - For a given category, the list of transactions is presented, ordered by the most recent.
      - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
      - The user fills out a form to create a new category, indicating their name and icon.
      - The user fills out a form to create a new transaction with:
      - name (mandatory)
      - amount (mandatory)
      - categories (mandatory at least one)
      - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
      - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- [ ] Testing requirements
      - Create unit and integration tests for all the most important components of your RoR application.

- [ ] Technical requirements
     -  You should use Postgres as your database
     - You should use devise for authentication.
     - You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
     - You can use a view template engine of your choice (.erb, .slim, .haml).
     - The project should be deployed and accessible online.

- [ ] Your database schema should reflect the following structure:

NOTE: do not change column names visible in the diagram. You need to change the "Entity" name according to the theme you have chosen for your project (please note: "Transaction" is a name already used by ActiveRecord, so using it as a name for your model and table will result in an error).

Project documentation
Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:

a description of the project.
a demo of the project features.
you should also highlight some interesting piece of code or something you built that you are very proud of.

### CODE REVIEW

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @mrkamin , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not use the proper readme template 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,2 +1,146 @@
...
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ](#faq)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 RoR-Budget-app <a name="about-project"></a>
```
Comment (Whoistolu) [inline README.md:29]: - [ ] [REQUIRED] A part of the project requirement states that you should record a video presenting the features of the project. The maximum length of the video is 5 minutes. Kindly ensure the video contains a clear and objective description of the project and the video should contain a demo of the project, highlighting all the project features. Lastly, the video should contain a highlight of a piece of the code you are proud of making.

- [ ] [REQUIRED] Also, ensure that the README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance] the author of the original design as required by the [Creative Commons license of the design.
- [ ] [REQUIRED] Kindly deploy the application using Render, Netlify or GitHub Pages as stated in the project requiremnt.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @mrkamin 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x]  _Check the inline comments under the review for sections to improve._
- [x] [REQUIRED] I am adding this recommendation her because we do not have the file in the file changes:

Your Implementation   |  Figma Design
:-------------------------:|:-------------------------:
Screenshot 2023-07-08 at 00 58 14 | Screenshot 2023-07-08 at 01 12 08

-  [x] Let's maintain the color of the navbar, and the New Category button provided on the Behance design. Also, ensure that you create a div for each category and flex the three items o each (Name, icon, and amount). Let's stick to the design provided on the Behance design.

- [ ] [REQUIRED] When I click on the create payment button, it throws an error that says a particular method is not defined. Kindly ensure that the method is defined.
Screenshot 2023-07-08 at 00 59 12


### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (mrkamin) [reply README.md:29]: @Whoistolu 
thanks for the good and valuable feedback,
I try my best to apply the suggestions you gave, 
Capture

File: README.md [hunk 2]
```diff
@@ -1,2 +1,146 @@
...
+
+If you like this project, Please give it a ⭐️⭐️
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+Thanks [@microverse](https://www.microverse.org/)
```
Comment (descholar-ceo) [inline README.md]: - [ ] The acknowledgment section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @mrkamin 🤝 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
👍🏽 Some of the comments have been addressed ✔️ 
👍🏽 No new issues introduced after addressing the comments ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] By clicking on the menu icon, the app is crashing with the error on the following screenshot
Screenshot 2023-07-08 at 14 23 56

- [ ] By clicking on the categories in the nabber, the app is crashing with the error on the screenshot
Screenshot 2023-07-08 at 14 27 22

- [ ] By clicking on next from the navbar while creating a transaction the app is crashing with the following error on the screenshot
Screenshot 2023-07-08 at 14 28 25

- [x] Transaction details page does not follow the given design from behance, kindly follow the design provided

| Your Design        | Behance's Design   |
| ------------------ | ------------------ |
| Screenshot 2023-07-08 at 14 46 10 | Screenshot 2023-07-08 at 14 45 52 |


- [x] As was required in the previous review, this comment wasn’t addressed yet: `[REQUIRED] A part of the project requirement states that you should record a video presenting the features of the project. The maximum length of the video is 5 minutes. Kindly ensure the video contains a clear and objective description of the project and the video should contain a demo of the project, highlighting all the project features. Lastly, the video should contain a highlight of a piece of the code you are proud of making.`
- [ ] As was requested in the previous review, the following comment wasn’t addressed yet: `[REQUIRED] Kindly deploy the application using Render, Netlify, or GitHub Pages as stated in the project requirements.`

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (mrkamin) [reply README.md]: @descholar-ceo 
thanks for your good and valuable feedback, 
all suggestions are appreciated, and applied on the projects.

Comment (mrkamin) [reply README.md]: Capture

Comment (mrkamin) [reply README.md]: video Presentation Link = ([https://www.loom.com/share/2826fc0f30414eebb472308d0c996977?sid=ed228c7c-c696-47a6-a84d-fdef3dde9474])

File: README.md [hunk 3]
```diff
@@ -1,2 +1,160 @@
...
+- Sign up and log in pages
+- Home page (categories page)
+- Transactions page
+- "Add a new category" page
+- "Add a new transaction" page
+- Testing requirements
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
```
Comment (uwadonat) [inline README.md:58]: - [ ] Your live demo link should be pointing to the application deployed using Render, Netlify, or github pages. Now it is pointing to your GitHub page. Kindly work on it and add the correct link.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,160 @@
...
+
+- [Live Demo Link]()
+
+## Project Presentation <a name="project-presentation"></a>
+
+- [Project Presentation video link](https://www.loom.com/share/2826fc0f30414eebb472308d0c996977?sid=cfeeffa6-6c65-4d94-8d06-0e9b65a6a264)
```
Comment (uwadonat) [inline README.md]: - [x] Thank you for adding the video and showing some of your code but some of the requirements are missing. Let's make sure you are showing all of the project features. here we can only see you showing the sign-up and sign-in parts.
scre

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @mrkamin , :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

### To highlight! :+1: 
- Descriptive pull request message ✔️
- No linter errors ✔️
- Well-organized and clean project  ✔️
almost

 ## Status: Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: README.md (hunk 4 above)
Comment (mrkamin) [reply README.md]: @uwadonat 
thanks for your good and valuable feedback, 
as the time is very limit 5 min so I have tried to explain all part of or required in this new video presentation 
https://www.loom.com/share/21dfbd82084a431baf0c54082f2e578d?sid=2b6d7b7e-3ed3-4d1d-8fe7-f0147ba17402
I hope I could covered all things this time

Comment (mrkamin) [reply README.md]: live demo = https://budgeat.onrender.com/

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @mrkamin  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Made changes as requested by the previous reviewer :+1: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

[OPTIONAL]
- [x] Please improve your microphone quality. As I was watching your presentation video, I had a hard time following through because it was difficult to pick the words.
Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

@Lucas-Masaba 
thanks for your good and valuable feedback,
sorry for making it hard for you it was because of time limit thanks once again I will consider your suggestion in the future

**Pull request merged**